src/syntax/abs/ast.rs

```diff
@@ -62,4 +63,6 @@ pub enum AbsDecl {
     Impl(SyntaxInfo, Abs),
     /// `Sign` and `Impl`
     Both(SyntaxInfo, Abs, SyntaxInfo, Abs),
+    /// Postulated value, like lambda parameters.
+    None(SyntaxInfo),
```

Comment:
In this way, `AbdDecl` becomes a tuple of two `Option`s. Consider replacing with a tuple.